p2d/p2d.py

```diff
@@ -114,55 +119,59 @@ def get_checker_md5(checker):
             if interactor is not None:
                 logger.info('Use custom interactor.')
                 with open(yaml_file, 'w', encoding='utf-8') as f:
+                    f.write(yaml_content)
                     f.write('validation: custom interactive\n')
                 ensure_dir(interactor_dir)
-                copyfile(testlib, '{}/testlib.h'.format(interactor_dir))
-                interactor_file = '{}/{}'.format(package_dir, interactor.find('source').attrib['path'])
-                copyfile(interactor_file, '{}/interactor.cpp'.format(interactor_dir))
+                copyfile(testlib, f'{interactor_dir}/testlib.h', follow_symlinks=False)
+                interactor_path = interactor.find('source').attrib['path']
+                interactor_file = f'{package_dir}/{interactor_path}'
+                copyfile(interactor_file, f'{interactor_dir}/interactor.cpp')
             elif checker is not None:
                 logger.info('Use custom checker.')
                 with open(yaml_file, 'w', encoding='utf-8') as f:
+                    f.write(yaml_content)
                     f.write('validation: custom\n')
                 ensure_dir(checker_dir)
-                copyfile(testlib, '{}/testlib.h'.format(checker_dir))
-                checker_file = '{}/{}'.format(package_dir, checker.find('source').attrib['path'])
-                copyfile(checker_file, '{}/checker.cpp'.format(checker_dir))
+                copyfile(testlib, f'{checker_dir}/testlib.h', follow_symlinks=False)
```

Comment:
What about marking `TESTLIB_PATH` as `"$CONFIG_PATH"/../` or `$(dirname "$CONFIG_PATH")`?

Reply:
have changed in python now~